packages/utils/src/download.ts

```diff
@@ -0,0 +1,43 @@
+import https from 'https';
+import fs from 'fs';
+
+/**
+ * Download the given URL to the given destination.
+ */
+export const downloadUrl = (url: string, dest: string): Promise<string> => {
+	return new Promise<string>((resolve, reject) => {
+		const file = fs.createWriteStream(dest, { flags: 'wx' });
+
+		const request = https.get(url, (response) => {
+			// make sure we have a valid status code
+			if (response.statusCode === 200) {
+				response.pipe(file);
+			} else {
+				file.close();
+				fs.unlink(dest, () => null);
+				reject(`Download failed! Server responded with ${response.statusCode}: ${response.statusMessage}`);
```

Comment:
you can also use `axios` and [`util.promisify`](https://nodejs.org/api/util.html#util_util_promisify_original) for `fs` to take advantage of async/await